src/common/setup.ts

```diff
@@ -0,0 +1,174 @@
+        `Warning: Your current kubernetes context (${runningContext}) does not match cluster context: ${process.env.K8S_CONTEXT}. Would you like to switch kube context to cluster first?`,
+        { defaultYes: true },
+      )
+    }
+    if (fixContext || (parser.argv as BasicArguments).setContext) {
+      await $`kubectl config use ${process.env.K8S_CONTEXT}`
+    }
+  }
+}
+/**
+ * Check the ENV_DIR parameter and whether or not the folder is populated
+ * @returns
+ */
+const checkEnvDir = (): boolean => {
+  if (dirname.includes('otomi-core') && !env.ENV_DIR) {
+    debug.checkEnvDir.exit(1, 'The ENV_DIR environment variable is not set')
+  }
+  debug.checkEnvDir.debug(env.ENV_DIR)
+  return readdirSync(env.ENV_DIR).length > 0
+}
+
+export type PrepareEnvironmentOptions = {
+  skipEnvDirCheck?: boolean
+  skipEvaluateSecrets?: boolean
+  skipKubeContextCheck?: boolean
+  skipDecrypt?: boolean
+  skipAll?: boolean
+}
+
+let clusterFile: any
+export const otomi = {
+  scriptName: process.env.OTOMI_CALLER_COMMAND ?? 'otomi',
+  /**
+   * Find the cluster kubernetes version in the values
+   * @returns String of the kubernetes version on the cluster
+   */
+  getK8sVersion: (): string => {
+    if (otomiK8sVersion) return otomiK8sVersion
+    if (!clusterFile) {
+      clusterFile = loadYaml(`${env.ENV_DIR}/env/cluster.yaml`)
+    }
+    otomiK8sVersion = clusterFile.cluster?.k8sVersion
+    return otomiK8sVersion
+  },
+  /**
+   * Find what image tag is defined in configuration for otomi
+   * @returns string
+   */
+  imageTag: (): string => {
+    if (otomiImageTag) return otomiImageTag
+    const file = `${env.ENV_DIR}/env/settings.yaml`
+    if (!existsSync(file)) return process.env.OTOMI_TAG ?? 'master'
+    const settingsFile = loadYaml(file)
+    otomiImageTag = settingsFile.otomi?.version ?? 'master'
+    return otomiImageTag
+  },
+  /**
+   * Find the customer name that is defined in configuration for otomi
+   * @returns string
+   */
+  customerName: (): string => {
+    if (otomiCustomerName) return otomiCustomerName
+    if (!clusterFile) {
+      clusterFile = loadYaml(`${env.ENV_DIR}/env/cluster.yaml`)
+    }
+    otomiCustomerName = clusterFile.cluster?.owner
+    return otomiCustomerName
+  },
+  /**
+   * Prepare environment when running an otomi command
+   */
+  prepareEnvironment: async (options?: PrepareEnvironmentOptions): Promise<void> => {
+    if (options && options.skipAll) return
+    debug.prepEnv.info('Checking environment')
+    if (!options?.skipEnvDirCheck && checkEnvDir()) {
+      if (!env.CI && !options?.skipEvaluateSecrets) await evaluateSecrets()
+      if (!env.CI && !options?.skipKubeContextCheck) await checkKubeContext()
+      if (!env.CI && !options?.skipDecrypt) await decrypt()
+    }
+  },
+  /**
+   * If ran within otomi-core, stop execution as it should not be ran within that folder.
+   * @param command that is executed
+   */
+  closeIfInCore: (command: string): void => {
```

Comment:
Perhaps `exitIfInCore` is better than `closeIfInCore`

Reply:
Makes sense!